### PR TITLE
Clean up & document

### PR DESCRIPTION
**Pull request by quarterto**
- remove memory check type that hasn't been usable for three years lol
- refactor to use async/await
- remove `next.` graphite check so it's usable outside of the program
- actually document it

this has ended up being a backwards compatible change ☺️ i'll be doing more drastic changes for v4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,97 +1,153 @@
 # n-health [![CircleCI](https://circleci.com/gh/Financial-Times/n-health.svg?style=svg)](https://circleci.com/gh/Financial-Times/n-health)
 
-Makes it easy to add a variety of healthchecks to an app.
+Collection of healthcheck classes to use in your nodejs application
```
Comment (adambraimbridge) [inline README.md]: Very nice change

File: README.md [hunk 2]
```diff
@@ -1,97 +1,153 @@
...
+* `checks`: Array of [check objects](#check-objects)
+
+### Check objects
+
+#### Common options
+
+* `type`: The type of check, which should be one of the types below. That check type's options should also be included in the object as required.
+* `name`, `severity`, `businessImpact`, `technicalSummary` and `panicGuide` are all required. See the [specification](https://docs.google.com/document/edit?id=1ftlkDj1SUXvKvKJGvoMoF1GnSUInCNPnNGomqTpJaFk) for details
+* `interval`: time between checks in milliseconds or any string compatible with [ms](https://www.npmjs.com/package/ms) [default: 1minute]
+* `officeHoursOnly`: [default: `false`] For queries that will probably fail out of hours (e.g. Internet Explorer usage, B2B stuff), set this to true and the check will pass on weekends and outside office hours. Use sparingly.
```
Comment (adambraimbridge) [inline README.md]:
```suggestion
* `officeHoursOnly`: [default: `false`] For queries that will probably fail out of hours (e.g. Internet Explorer usage, B2B stuff), set this to true and the check will pass on weekends and outside BST (British Standard Time) office hours. Use sparingly.
```

File: src/checks/responseCompare.check.js [hunk 3]
```diff
@@ -34,25 +34,28 @@ class ResponseCompareCheck extends Check {
...
-			.catch(err => {
-				console.error(err.stack);
-				setTimeout(() => {throw err; }, 0);
-			});
+	async tick(){
+		try {
+			const responses = await Promise.all(
+				this.urls.map(
+					url => fetch(url, { headers: this.headers }).then(
+						r => r.text().then(this.normalizeResponse)
```
Comment (adambraimbridge) [inline src/checks/responseCompare.check.js]: `fetchres.text` perhaps, for consistency?

Comment (quarterto) [reply src/checks/responseCompare.check.js]: not quite equivalent, because `fetchres` is also checking `res.ok`

**Pull request merged**